shpc/main/modules/base.py

```diff
@@ -338,30 +338,29 @@ def check(self, module_name):
         at updates for entire tags. If a specific folder is provided with
         a container, check the digest.
         """
-        module = self.new_module(module_name)
+        module = self.new_module(module_name, in_registry=True)
```

Comment:
What about having a `get_module` function that basically calls `new_module` with in_registry set to True? That way, the developer reading this line isn't conflicted between the naming - the idea of a new module contrasted with already being in the registry.